api/PclusterApiHandler.py

```diff
@@ -130,7 +148,13 @@ def authenticate(groups):
     try:
         decoded = jwt_decode(access_token)
     except jwt.ExpiredSignatureError:
-        return abort(401)
+        refresh_token = request.cookies.get('refreshToken', None)
+        if refresh_token is None:
+            return abort(401)
```

Comment:
suggestion: I believe this case is already covered by the `refresh_tokens` function (I'm guessing it will raise a `RefreshTokenError`)

Reply:
As it is right now, refresh_tokens performs the POST and expects a refresh token
I preferred to leave the concern of checking for the presence of the token outside the function that uses it